dms/security/security.xml

```diff
@@ -68,34 +81,109 @@
         <field name="perm_unlink" eval="1" />
         <field name="domain_force">[(1 ,'=', 1)]</field>
     </record>
-    <record id="rule_security_dms_directory" model="ir.rule">
-        <field
-            name="name"
-        >Dms Directories with dms group (will be filtered later by code)</field>
-        <field name="model_id" ref="model_dms_directory" />
+    <!-- Forbid lower groups access to hidden storage -->
+    <record id="rule_forbid_hidden_storage" model="ir.rule">
+        <field name="name">Basic users cannot access hidden storage</field>
+        <field name="model_id" ref="model_dms_storage" />
         <field
             name="groups"
-            eval="[(4, ref('base.group_user')),(4, ref('base.group_portal'))]"
+            eval="[(4, ref('base.group_portal')), (4, ref('group_dms_user'))]"
         />
         <field name="perm_read" eval="1" />
+        <field name="perm_create" eval="1" />
+        <field name="perm_write" eval="1" />
+        <field name="perm_unlink" eval="1" />
+        <field name="domain_force">[('is_hidden', '=', False)]</field>
+    </record>
+    <record id="rule_allow_hidden_storage" model="ir.rule">
+        <field name="name">Managers can access hidden storage</field>
+        <field name="model_id" ref="model_dms_storage" />
+        <field name="groups" eval="[(4, ref('group_dms_manager'))]" />
+        <field name="perm_read" eval="1" />
+        <field name="perm_create" eval="1" />
+        <field name="perm_write" eval="1" />
+        <field name="perm_unlink" eval="1" />
+        <field name="domain_force">[('is_hidden', '=', True)]</field>
+    </record>
+    <!-- These rules leverage computed permission management -->
+    <record id="rule_directory_computed_create" model="ir.rule">
+        <field name="name">Apply computed create permissions.</field>
+        <field name="model_id" ref="model_dms_directory" />
+        <field name="global" eval="True" />
+        <field name="perm_read" eval="0" />
+        <field name="perm_create" eval="1" />
+        <field name="perm_write" eval="0" />
+        <field name="perm_unlink" eval="0" />
+        <field name="domain_force">[('permission_create', '=', True)]</field>
```

Comment:
And the last HACK from v12 also removed: https://github.com/OCA/dms/blob/db9cb933ba4f29cc1a610190c32e2ff4bb16b1ce/dms/security/security.xml#L119

Nice!